lib/salus/report.rb

```diff
@@ -222,9 +224,26 @@ def publish_report(directive)
       end
     end
 
+    def satisfies_filter?(directive)
+      return true if @report_filter == 'all'
+      return false if @report_filter == 'none'
+
+      # Use regex to capture two parts of the filter, delimited by a ':'
+      filter_regex = Regexp.new('([a-zA-Z0-9]+):?([a-zA-Z0-9\*]*)')
```

Comment:
It doesn't seem necessary to use a regex.
I think all you have to do is something like
```ruby
key, value = @report_filter.split(':')
```

Reply:
I went with a regex, because it's a more natural approach to wildcard filter. Also, I can change the regex to support all `-` in addition to other special characters with something like `([a-zA-Z0-9-_\\\.]+):?([a-zA-Z0-9\*]*)`.

Reply:
`-` is just an example, we may allow other characters too.

@joshua-ostrom  Any thoughts on this?  This is just breaking like
`format: abcd` into key `format` and value `abcd`.